src/main.cpp

```diff
@@ -242,29 +243,45 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
     Napi::Error::New(env, "Calling run on a disposed context").ThrowAsJavaScriptException();
     return env.Null();
   }
+
   if (info.Length() < 2) {  // inputs, timeout
-    Napi::Error::New(env, "Wrong number of arguments, expected 2").ThrowAsJavaScriptException();
+    Napi::Error::New(env, "Wrong number of arguments, at least 2 expected").ThrowAsJavaScriptException();
     return env.Null();
   }
-  if (!info[0].IsObject()) {
-    Napi::TypeError::New(env, "First argument must be an object").ThrowAsJavaScriptException();
+
+  if (!info[0].IsObject() && !info[1].IsObject()) {
+    Napi::TypeError::New(env, "One of persistent data or ephemeral data must be an object").ThrowAsJavaScriptException();
     return env.Null();
   }
-  if (!info[1].IsNumber()) {
-    Napi::TypeError::New(env, "Second argument must be a number").ThrowAsJavaScriptException();
+
+  if (!info[2].IsNumber()) {
```

Comment:
What if `info.Length() == 2`? Isn't this going to be a segmentation fault?

Reply:
Yep, I overlooked that.